osquery/extensions/impl_thrift.cpp

```diff
@@ -369,8 +371,10 @@ void ExtensionRunnerInterface::init(RouteUUID uuid, bool manager) {
         std::make_shared<extensions::ExtensionManagerProcessor>(handler);
   }
   // Set the global output function for thrift
-  GlobalOutput.setOutputFunction(
-      [](const char* message) -> void { VLOG(1) << "Thrift: " << message; });
+  if (FLAGS_thrift_verbose) {
+    GlobalOutput.setOutputFunction(
```

Comment:
I am curious what happens when we do not call `setOutputFunction`? Is it the case that there is a default, thrift-provided, output function?

Do we need to set this to a no-op when our flag is not enabled?

Reply:
So the default is set [here](https://github.com/apache/thrift/blob/077b5fce825e79d84592fff893639b92b637eec7/lib/cpp/src/thrift/TOutput.cpp#L36) and ends up calling:
https://github.com/apache/thrift/blob/077b5fce825e79d84592fff893639b92b637eec7/lib/cpp/src/thrift/TOutput.cpp#L92

Which will cause it to print out to the stderr. I thought that the thrift [CmakeList.txt](https://github.com/osquery/osquery/blob/master/libraries/cmake/source/thrift/CMakeLists.txt) had defined ```THRIFT_SQUELCH_CONSOLE_OUTPUT``` at one point but maybe I've seen that somewhere else. Either we define it, accept the prints (which I think [this](https://github.com/osquery/osquery/pull/6592) was originally removing) I can't see a ``` if(!f_)``` so we'd probably have to do something like 
```
GlobalOutput.setOutputFunction(
      [](const char* message) -> void {});
```
to remove the call.

Reply:
Ah cool, thanks for researching and confirming. Right, this change should add that no-op lambda you recommend if the flag is not enabled.

Reply:
Done